db/migrate/20230525200407_fix_vpc_provision_instance_type.rb

```diff
@@ -0,0 +1,42 @@
+class FixVpcProvisionInstanceType < ActiveRecord::Migration[6.1]
+  class VmOrTemplate < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+
+    self.inheritance_column = :_type_disabled
+    self.table_name = "vms"
+  end
+
+  class MiqRequest < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    say_with_time("Setting instance_type for VPC provision requests") do
+      MiqRequest.in_my_region.where(
+        :type   => "MiqProvisionRequestTemplate",
+        :source => VmOrTemplate.where(:type => "ManageIQ::Providers::IbmCloud::VPC::CloudManager::Template")
```

Comment:
I guess you need to define the source association in your MiqRequest stub or build up the list of ids and query for `source_id` in [ids] and `source_type => VmOrTemplate`

Reply:
```
StandardError: An error has occurred, this and all later migrations canceled:

PG::UndefinedColumn: ERROR:  column miq_requests.source does not exist
LINE 1: ...WEEN $1 AND $2 AND "miq_requests"."type" = $3 AND "miq_reque...
                                                             ^
HINT:  Perhaps you meant to reference the column "miq_requests.source_id".
```